sphinx_gallery/gen_rst.py

```diff
@@ -439,8 +439,13 @@ def generate_dir_rst(src_dir, target_dir, gallery_conf, seen_backrefs):
 
     if not os.path.exists(target_dir):
         os.makedirs(target_dir)
+    # get filenames
     listdir = [fname for fname in os.listdir(src_dir)
                if fname.endswith('.py')]
+    # limit which to look at based on regex
+    listdir = [fname for fname in listdir
+               if re.match(gallery_conf['ignore_pattern'], fname) is None]
```

Comment:
As far as I remember, filename_pattern uses the full path. It would be great if ignore_pattern was using the full path as well.

For the same reason, I would use `re.search` rather than `re.match`.